Naming.md

```diff
@@ -6,7 +6,15 @@ In general, follow the [official guidelines](https://swift.org/documentation/api
 
 ### Naming a file
 
-There's no three-letter prefix in Swift file naming. For example, if an Objective-C file named XYZEvent.m is to be converted to Swift, it should be renamed to Event.swift.
+- All Swift files should use the `.swift` extension.
+- The name of a file should make it clear what the contents are. For example, if there is a single class named `EasyTapButton` then the file should be called `EasyTapButton.swift`. If you have a view controller, along with some helper types, then using the name of the view controller will likely be the best choice. A good heuristic to use is that if it's ever unclear what to name a file, it may be worth splitting that file up.
+- There's no three-letter prefix in Swift file naming. For example, if an Objective-C file named XYZEvent.m is to be converted to Swift, it should be renamed to Event.swift.
+- If a file contains only one or more extensions, then the file name should be `MyType+Extensions.swift` unless the extension have a specific name; in that case the extension should be named `MyType+ExtensionName.swift`.
```

Comment:
I was under the impression that Swift extensions were always unnamed?

From [Extensions — The Swift Programming Language (Swift 5.5)](https://docs.swift.org/swift-book/LanguageGuide/Extensions.html):
> (Unlike Objective-C categories, Swift extensions don’t have names.)

Reply:
Thanks for pointing that out! Will fix in next iteration.

Reply:
While this is mostly true, if the extension is purely adopting a specific protocol it might be worth having an exception for having it be `MyType+ProtocolBeingConformedTo.swift`. Like `MyType+NSCopying.swift` feels more useful if that's what the file is doing. Thoughts?

Reply:
I'm 100% on board with sticking to the `MyType+Protocol.swift` or even `MyType+WhatWouldTheExtensionBeNamedIfItWereObjectiveC.swift`. Sorry I didn't have more detail in my comment! I meant to point out that we just need to update the phrasing of this particular rule.

Reply:
minor: `unless the extension have a specific name` should be `unless the extension **has** a specific name`

Reply:
Yeah, what Mike said is what our intent here is. Usually we'd use the protocol name if there's only one and that's what the extension is for. If there isn't a protocol, then it's just a matter of finding a sensible name for it.